test/cpp/api/dataloader.cpp

```diff
@@ -792,3 +783,44 @@ TEST(DataLoaderTest, TestExceptionsArePropagatedFromWorkers) {
         std::rethrow_exception(e.original_exception), std::invalid_argument);
   }
 }
+
+TEST(DataTest, DataLoaderWithChunkSupportSingleWorker) {
+  const size_t kBatchSize = 13;  
+
+  auto dataset = InfiniteStreamDataset().map(
+      transforms::Lambda<int>([](int x) { return x + 1; }));
+
+  auto data_loader = torch::data::make_chunk_data_loader(
+      std::move(dataset),
+      DataLoaderOptions().batch_size(kBatchSize),
+      samplers::BatchSizeSampler{});
+
+  auto iterator = data_loader->begin();
+  for (size_t i = 0; i < 3; ++i, ++iterator) {
+    ASSERT_NE(iterator, data_loader->end());
+    std::vector<int> batch = *iterator;
+    ASSERT_EQ(batch.size(), kBatchSize);
+    for (size_t j = 0; j < kBatchSize; ++j) {
+      ASSERT_EQ(batch.at(j), 1 + (i * kBatchSize) + j);
+    }
+  }
+}
+
+TEST(DataTest, DataLoaderWithChunkSupportMultiWorkers) {
+  const size_t kBatchSize = 13;  
+  
+  InfiniteStreamDataset dataset; 
+
+  auto data_loader = torch::data::make_chunk_data_loader(
+      std::move(dataset),
+      DataLoaderOptions().batch_size(kBatchSize).workers(3),
+      samplers::BatchSizeSampler{});
+
+  auto iterator = data_loader->begin();
+  for (size_t i = 0; i < 3; ++i, ++iterator) {
+    ASSERT_NE(iterator, data_loader->end());
+    std::vector<int> batch = *iterator;
+    ASSERT_EQ(batch.size(), kBatchSize);   
+  }
```

Comment:
Do we also need to compare the content in each batch, like the case in DataLoaderWithChunkSupportSingleWorker?

Reply:
We could, but the ordering is not guaranteed with threads. So I just leave it at the number of elements. We will add more tests with the chunk data set implementation.

Reply:
The order should be managed by sequencer_ inside data loader. By default it is OrderedSequencer which returns the batch in order. Or maybe I miss anything here?

Reply:
You are right, I thought by default we have a NoSequencer, which seems like the approach I would prefer, but with a sequencer we can test for the contents.  Let me explicitly add the option and check for the content.